From 39e1c4aa9c1bfeda4c23848ec8f0a1a37c4cbfbc Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 18 Sep 2023 16:39:33 +0100
Subject: [PATCH] overlays: Correct for #pwm-cells = 3

An upstream commit changed #pwm-cells for the BCM2835 PWMs to 3, so as
to accomodate the polarity. Add a zero to all references to the PWMs
to indicate normal polarity.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/cutiepi-panel-overlay.dts     | 2 +-
 arch/arm/boot/dts/overlays/mipi-dbi-spi-overlay.dts      | 2 +-
 arch/arm/boot/dts/overlays/pwm-ir-tx-overlay.dts         | 2 +-
 arch/arm/boot/dts/overlays/vc4-kms-dpi.dtsi              | 2 +-
 arch/arm/boot/dts/overlays/watterott-display-overlay.dts | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/overlays/cutiepi-panel-overlay.dts b/arch/arm/boot/dts/overlays/cutiepi-panel-overlay.dts
index 6f9694e81d6a..d14c3698eb75 100644
--- a/arch/arm/boot/dts/overlays/cutiepi-panel-overlay.dts
+++ b/arch/arm/boot/dts/overlays/cutiepi-panel-overlay.dts
@@ -60,7 +60,7 @@ rpi_backlight: rpi_backlight {
                 compatible = "pwm-backlight";
                 brightness-levels = <0 6 8 12 16 24 32 40 48 64 96 128 160 192 224 255>;
                 default-brightness-level = <6>;
-                pwms = <&pwm 0 200000>;
+                pwms = <&pwm 0 200000 0>;
                 power-supply = <&vdd_3v3_reg>;
                 status = "okay";
             };
diff --git a/arch/arm/boot/dts/overlays/mipi-dbi-spi-overlay.dts b/arch/arm/boot/dts/overlays/mipi-dbi-spi-overlay.dts
index 040f76dffc6c..63fb3a5f2388 100644
--- a/arch/arm/boot/dts/overlays/mipi-dbi-spi-overlay.dts
+++ b/arch/arm/boot/dts/overlays/mipi-dbi-spi-overlay.dts
@@ -81,7 +81,7 @@ backlight_pwm: backlight_pwm {
 				compatible = "pwm-backlight";
 				brightness-levels = <0 6 8 12 16 24 32 40 48 64 96 128 160 192 224 255>;
 				default-brightness-level = <15>;
-				pwms = <&pwm 0 200000>;
+				pwms = <&pwm 0 200000 0>;
 			};
 		};
 	};
diff --git a/arch/arm/boot/dts/overlays/pwm-ir-tx-overlay.dts b/arch/arm/boot/dts/overlays/pwm-ir-tx-overlay.dts
index 119caf746b3b..33597eb79729 100644
--- a/arch/arm/boot/dts/overlays/pwm-ir-tx-overlay.dts
+++ b/arch/arm/boot/dts/overlays/pwm-ir-tx-overlay.dts
@@ -28,7 +28,7 @@ fragment@2 {
 		__overlay__ {
 			pwm-ir-transmitter {
 				compatible = "pwm-ir-tx";
-				pwms = <&pwm 0 100>;
+				pwms = <&pwm 0 100 0>;
 			};
 		};
 	};
diff --git a/arch/arm/boot/dts/overlays/vc4-kms-dpi.dtsi b/arch/arm/boot/dts/overlays/vc4-kms-dpi.dtsi
index 27bb76598701..67c884de2a8d 100644
--- a/arch/arm/boot/dts/overlays/vc4-kms-dpi.dtsi
+++ b/arch/arm/boot/dts/overlays/vc4-kms-dpi.dtsi
@@ -66,7 +66,7 @@ backlight_pwm: backlight_pwm {
 				compatible = "pwm-backlight";
 				brightness-levels = <0 6 8 12 16 24 32 40 48 64 96 128 160 192 224 255>;
 				default-brightness-level = <16>;
-				pwms = <&pwm 0 200000>;
+				pwms = <&pwm 0 200000 0>;
 			};
 		};
 	};
diff --git a/arch/arm/boot/dts/overlays/watterott-display-overlay.dts b/arch/arm/boot/dts/overlays/watterott-display-overlay.dts
index c0e20afa3bf0..4388706d2c38 100644
--- a/arch/arm/boot/dts/overlays/watterott-display-overlay.dts
+++ b/arch/arm/boot/dts/overlays/watterott-display-overlay.dts
@@ -110,7 +110,7 @@ backlight_pwm: backlight_pwm {
 				compatible = "pwm-backlight";
 				brightness-levels = <0 6 8 12 16 24 32 40 48 64 96 128 160 192 224 255>;
 				default-brightness-level = <16>;
-				pwms = <&pwm 0 200000>;
+				pwms = <&pwm 0 200000 0>;
 			};
 		};
 	};
-- 
2.34.1

